src/plugins/mission_raw/mission_raw_impl.cpp

```diff
@@ -87,40 +201,24 @@ MAVLinkMissionTransfer::ItemInt MissionRawImpl::convert_mission_raw(
     return new_item_int;
 }
 
-MissionRaw::Result MissionRawImpl::convert_result(MAVLinkMissionTransfer::Result result)
+std::vector<MAVLinkMissionTransfer::ItemInt> MissionRawImpl::convert_to_int_items(
+    const std::vector<std::shared_ptr<MissionRaw::MavlinkMissionItemInt>>& mission_raw)
 {
-    switch (result) {
-        case MAVLinkMissionTransfer::Result::Success:
-            return MissionRaw::Result::SUCCESS;
-        case MAVLinkMissionTransfer::Result::ConnectionError:
-            return MissionRaw::Result::ERROR; // FIXME
-        case MAVLinkMissionTransfer::Result::Denied:
-            return MissionRaw::Result::ERROR; // FIXME
-        case MAVLinkMissionTransfer::Result::TooManyMissionItems:
-            return MissionRaw::Result::ERROR; // FIXME
-        case MAVLinkMissionTransfer::Result::Timeout:
-            return MissionRaw::Result::TIMEOUT;
-        case MAVLinkMissionTransfer::Result::Unsupported:
-            return MissionRaw::Result::UNSUPPORTED;
-        case MAVLinkMissionTransfer::Result::UnsupportedFrame:
-            return MissionRaw::Result::ERROR; // FIXME
-        case MAVLinkMissionTransfer::Result::NoMissionAvailable:
-            return MissionRaw::Result::NO_MISSION_AVAILABLE;
-        case MAVLinkMissionTransfer::Result::Cancelled:
-            return MissionRaw::Result::CANCELLED;
-        case MAVLinkMissionTransfer::Result::MissionTypeNotConsistent:
-            return MissionRaw::Result::INVALID_ARGUMENT; // FIXME
-        case MAVLinkMissionTransfer::Result::InvalidSequence:
-            return MissionRaw::Result::INVALID_ARGUMENT; // FIXME
-        case MAVLinkMissionTransfer::Result::CurrentInvalid:
-            return MissionRaw::Result::INVALID_ARGUMENT; // FIXME
-        case MAVLinkMissionTransfer::Result::ProtocolError:
-            return MissionRaw::Result::ERROR; // FIXME
-        case MAVLinkMissionTransfer::Result::InvalidParam:
-            return MissionRaw::Result::INVALID_ARGUMENT; // FIXME
-        default:
-            return MissionRaw::Result::UNKNOWN;
+    std::vector<MAVLinkMissionTransfer::ItemInt> int_items;
+
+    // FIXME item_i only increment when new NAV_WAYPOINT
+    int item_i = -1;
+
+    for (const auto& item : mission_raw) {
+        if (item->command == MAV_CMD_NAV_WAYPOINT) {
+            ++item_i;
```

Comment:
I think this logic is not needed. For current I would just use the actual MAVLink index. The map like for `mission` is not required here, right?